endpoint/endpoint.go

```diff
@@ -23,3 +23,11 @@ type Endpoint struct {
 	// The target the DNS record points to
 	Target string
 }
+
+// SharedEndpoint is a unit of data stored in the storage it should provide information such as
+// 1. Owner - which external-dns instance is managing the records
+// 2. DNSName and Target inherited from endpoint.Endpoint struct
+type SharedEndpoint struct {
+	Owner string //refers to the Owner ID
+	Endpoint
+}
```

Comment:
I had some ideas regarding this: Ideally we want to have the DNS provider to allow attaching labels to DNS records, so that we can model our ownership on top of that. Correct me if i'm wrong.

However, Google/AWS don't have that but that doesn't mean we cannot expose it this way to ExternalDNS. We could find a DNSRecord/Endpoint abstraction that looks like this:

```go
type DNSRecord struct {
  Source, Target string
  Labels map[string]string
}
```

and then treat records as if this were a native feature of the DNS provider (we would have to do the translation to configMap or TXT records somewhere for providers that don't allow to do this more natively)

The `SharedEndpoint struct` encapsulates exactly the same idea just with some different naming (fixed owner instead of arbitrary labels). So the direction is good but I would like to discuss the naming and usage a little more.